### PR TITLE
[13.0][ADD] sale_coupon_multiple_code_program: New module

### PR DESCRIPTION
**Pull request by chienandalu**
Coupon Programs only allow one code promotion program per sale. There's no technical or practical reason to disallow it.

To use this module, you need to:

1. Go to *Sales > Master Data > Coupon Programs*
1. Configure several programs with their respective program codes.
1. Place a new order that fulfill such programs requisites.
1. Apply the programs codes.
1. You'll get every promotion applied.

cc @Tecnativa TT38413

please review @CarlosRoca13 @victoralmau

### CODE REVIEW

**Review by victoralmau — APPROVED** [review]
Code review OK.

File: sale_coupon_limit/models/sale_coupon_program.py [hunk 1]
```diff
@@ -15,11 +15,13 @@ def _check_promo_code(self, order, coupon_code):
...
-            [("promo_code", "=", coupon_code)]
-            if coupon_code
-            else [("no_code_promo_program_ids", "in", self.ids)]
-        )
+        domain = [
+            ("id", "!=", order._origin.id),
+            ("state", "!=", "cancel"),
+            "|",
+            ("no_code_promo_program_ids", "in", self.ids),
+            ("code_promo_program_id", "=", self.id),
```
Comment (pedrobaeza) [inline sale_coupon_limit/models/sale_coupon_program.py:23]: no `coupon_code` check now?

Comment (pedrobaeza) [reply sale_coupon_limit/models/sale_coupon_program.py:23]: @chienandalu can you answer this?

Comment (chienandalu) [reply sale_coupon_limit/models/sale_coupon_program.py:23]: Well, `promo_code` is just a relation to the correspondant `code_promo_program_id` record.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge patch

**Pull request merged**